src/EditorFeatures/CSharpTest/Completion/CompletionProviders/ObjectCreationCompletionProviderTests.cs

```diff
@@ -438,6 +439,58 @@ void M1()
             await VerifyItemExistsAsync(markup, "object");
         }
 
+        [Fact, Trait(Traits.Feature, Traits.Features.Completion)]
+        [WorkItem(4585, "https://github.com/dotnet/roslyn/issues/4585")]
+        public async Task GenericArguments_PredefinedType()
+        {
+            var markup = @"
+using System;
+
+class A<T> { }
+
+class B
+{
+    A<Int32> _a = new $$
+}";
+
+            await VerifyItemExistsAsync(markup, "A<int>",
+                options: Option(CodeStyleOptions.PreferIntrinsicPredefinedTypeKeywordInDeclaration, LanguageNames.CSharp, true));
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.Completion)]
+        [WorkItem(4585, "https://github.com/dotnet/roslyn/issues/4585")]
+        public async Task GenericArguments_FrameworkType()
+        {
+            var markup = @"
+using System;
+
+class A<T> { }
+
+class B
+{
+    A<int> _a = new $$
+}";
+
+            await VerifyItemExistsAsync(markup, "A<Int32>",
+                options: Option(CodeStyleOptions.PreferIntrinsicPredefinedTypeKeywordInDeclaration, LanguageNames.CSharp, false));
```

Comment:
I don't quite understand this. Why do we need to have options involved in completion at all? This doesn't seem like a good behavior to me. I would expect this to respect what you actually typed in the declaration as opposed to checking your style preferecnes:
```c#
A<int> a = new $$ // I would expect A<int> to show up here
A<Int32>a = new && // I would expect A<Int32> to show up here
```

Reply:
Today it doesn't respect either what I type _or_ the preferences I've set. In most cases I would expect the desire is to have both the variable and creation styles match the current preferences.

Taking it further would probably require a good chunk of extra work for the minimizing process to produce the correct result, and dealing with cases where the target of the assignment doesn't have any available source code, or where that source code style preferences don't match the current document. Not something I'd consider worth the effort.

Reply:
My problem with this is that completion is forcing you to a style change even though you already explicitly typed what you typed in a different style for whatever reason. Completion should respect that first.

>  I would expect the desire is to have both the variable

Then write it that way in the first place :smiley: If you don't, completion shouldn't be forcing you to.

Reply:
> where the target of the assignment doesn't have any available source code

could you clarify what you mean by that?

Reply:
@CyrusNajmabadi what do you think about this?

Reply:
THere are two situations to care about:

1. ```some_expr = new$$```  and
2. ```ExplicitType t = new $$```

Yes.  In the second case, it seems reasonable to probably just suggest whatever type was explicit provided on the left.  This code, however, is clearly trying to make sure that case 1 works properly.  i.e. if you had an expression that imported a type (say ```List<int>```) that "new" will respect your wishes.

Reply:
> I don't quite understand this. Why do we need to have options involved in completion at all? 

Because sometimes you have types that you don't know how they were originally declared (this is common) and you want to ensure that any generation you do follows the style that hte user has explicitly stated they want.